src/main/kotlin/com/avail/AvailRuntime.kt

```diff
@@ -0,0 +1,1814 @@
+			val method: A_Method = bundle.bundleMethod()
+			method.removeSealedArgumentsType(sealSignature)
+		}
+		finally
+		{
+			runtimeLock.writeLock().unlock()
+		}
+	}
+
+	/**
+	 * The [lock][ReentrantLock] that guards access to the Level One
+	 * [-safe][levelOneSafeTasks] and [-unsafe][levelOneUnsafeTasks] queues and
+	 * counters.
+	 *
+	 * For example, an [L2Chunk] may not be [invalidated][L2Chunk.invalidate]
+	 * while any [fiber][FiberDescriptor] is
+	 * [running][FiberDescriptor.ExecutionState.RUNNING] a Level Two chunk.
+	 * These two activities are mutually exclusive.
+	 */
+	private val levelOneSafeLock = ReentrantLock()
+
+	/**
+	 * The [queue][Queue] of Level One-safe [tasks][Runnable]. A Level One-safe
+	 * task requires that no
+	 * [Level&#32;One-unsafe&#32;tasks][levelOneUnsafeTasks] are running.
+	 *
+	 * For example, a [Level Two chunk][L2Chunk] may not be
+	 * [invalidated][L2Chunk.invalidate] while any [fiber][FiberDescriptor] is
+	 * [running][FiberDescriptor.ExecutionState.RUNNING] a Level Two chunk.
+	 * These two activities are mutually exclusive.
+	 */
+	private val levelOneSafeTasks: Queue<AvailTask> = ArrayDeque()
+
+	/**
+	 * The [queue][Queue] of Level One-unsafe [tasks][Runnable]. A Level
+	 * One-unsafe task requires that no [Level One-safe
+	 * tasks][levelOneSafeTasks] are running.
+	 */
+	private val levelOneUnsafeTasks: Queue<AvailTask> = ArrayDeque()
+
+	/**
+	 * The number of [Level&#32;One-safe&#32;tasks][levelOneSafeTasks] that have
+	 * been [scheduled&#32;for&#32;execution][executor] but have not yet reached
+	 * completion.
+	 */
+	private var incompleteLevelOneSafeTasks = 0
+
+	/**
+	 * The number of [Level&#32;One-unsafe&#32;tasks][levelOneUnsafeTasks] that
+	 * have been [scheduled&#32;for&#32;execution][executor] but have not yet
+	 * reached completion.
+	 */
+	private var incompleteLevelOneUnsafeTasks = 0
+
+	/**
+	 * Has [whenLevelOneUnsafeDo] Level One safety been requested?
+	 */
+	@Volatile
+	private var levelOneSafetyRequested = false
+
+	/**
+	 * Has [whenLevelOneUnsafeDo] Level One safety} been requested?
+	 *
+	 * @return
+	 *   `true` if Level One safety has been requested, `false` otherwise.
+	 */
+	fun levelOneSafetyRequested(): Boolean =  levelOneSafetyRequested
+
+	/**
+	 * Request that the specified continuation be executed as a Level One-unsafe
+	 * task at such a time as there are no Level One-safe tasks running.
+	 *
+	 * @param priority
+	 *   The priority of the [AvailTask] to queue.  It must be in the range
+	 *   [0..255].
+	 * @param unsafeAction
+	 *   The action to perform when Level One safety is not required.
+	 */
+	fun whenLevelOneUnsafeDo(priority: Int, unsafeAction: () -> Unit)
+	{
+		val wrapped = AvailTask(priority)
+		{
+			try
+			{
+				unsafeAction.invoke()
+			}
+			catch (e: Exception)
+			{
+				System.err.println(
+					"\n\tException in level-one-unsafe task:\n\t${trace(e)}"
+						.trimIndent())
+			}
+			finally
+			{
+				levelOneSafeLock.lock()
```

Comment:
This one requires `levelOneSafeLock.withLock { ... }`.

Reply:
I presume the code inside the try block should be moved into the withLock block?